onpremise/onpremise.go

```diff
@@ -225,14 +314,16 @@ type Evidence struct {
 	Value  string
 }
 
-func (p *Engine) Process(evidences []Evidence) (*dd.ResultsHash, error) {
+func (e *Engine) Process(evidences []Evidence) (*dd.ResultsHash, error) {
+	e.RLock()
```

Comment:
so we can not use locks, because it is a performance disaster

Reply:
the solution is to not recreate the manager, but always reload from file: 
* either using `HashReloadManagerFromOriginalFile` - that function we have exposed in Go
* or `HashReloadManagerFromFile`  - this we don't have exposed - we'll need to expose it - please add a wrapper for it to the https://github.com/51Degrees/device-detection-go/blob/main/dd/resource_hash.go - in a similar way it has `C.HashReloadManagerFromOriginalFile`